Userland/Libraries/LibPDF/ColorSpace.cpp

```diff
@@ -687,16 +687,70 @@ Vector<float> IndexedColorSpace::default_decode() const
     return { 0.0, static_cast<float>(m_hival) };
 }
 
-PDFErrorOr<NonnullRefPtr<SeparationColorSpace>> SeparationColorSpace::create(Document*, Vector<Value>&&)
+PDFErrorOr<NonnullRefPtr<SeparationColorSpace>> SeparationColorSpace::create(Document* document, Vector<Value>&& parameters)
 {
-    auto color_space = adopt_ref(*new SeparationColorSpace());
-    // FIXME: Implement.
+    if (parameters.size() != 3)
+        return Error { Error::Type::MalformedPDF, "Separation color space expected three parameters" };
+
+    // "The name parameter is a name object specifying the name of the colorant that this Separation color space
+    //  is intended to represent (or one of the special names All or None; see below)"
+    auto param0 = TRY(document->resolve(parameters[0]));
+    if (!param0.has<NonnullRefPtr<Object>>())
+        return Error { Error::Type::MalformedPDF, "Separation color space expects object for first arg" };
+    auto name_object = param0.get<NonnullRefPtr<Object>>();
+    if (!name_object->is<NameObject>())
+        return Error { Error::Type::MalformedPDF, "Separation color space expects name object for first arg" };
```

Comment:
Would it make sense to use a `TRY` pattern here?
Something like
```cpp
auto name_object = TRY(param0.get_if<NonnullRefPtr<Object>>());
```

Reply:
Maybe even some kinda
```cpp
auto name_object = TRY(param0.get_if<NameObject>());
```

Reply:
Belated: #21921